### PR TITLE
Intrinsics unchecked ops

### PR DESCRIPTION
**Pull request by CohenArthur**
Addresses #658

### CODE REVIEW

File: gcc/rust/backend/rust-compile-intrinsic.cc [hunk 1]
```diff
@@ -34,6 +34,22 @@
 namespace Rust {
 namespace Compile {
 
+static bool
+is_basic_integer_type (TyTy::BaseType *type)
+{
+  switch (type->get_kind ())
+    {
+    case TyTy::INT:
+    case TyTy::UINT:
```
Comment (bjorn3) [inline gcc/rust/backend/rust-compile-intrinsic.cc:43]: Do these cover all u* and i* sizes other than usize and isize?

Comment (CohenArthur) [reply gcc/rust/backend/rust-compile-intrinsic.cc:43]: Yes, they do. This is similar to [`rustc`'s`TyKind` enum](https://doc.rust-lang.org/stable/nightly-rustc/rustc_middle/ty/enum.TyKind.html) except we can't have sum types, so we're using inheritance to counter that. All of our `TyTy` nodes inherit from the `BaseType` class, which exposes `get_kind()`. Then specifically the `IntType` and `UintType` classes (which inherit from `BaseType`) expose one new enum each and a `get_{u}int_kind()` method

**Review by dafaust — APPROVED** [review]
Cool, LGTM

**Pull request merged**